### PR TITLE
Write project guide

### PR DESCRIPTION
**Pull request by ibraheemdev**
## Summary

Write the project guide that was added in https://github.com/astral-sh/uv/pull/5135.

I tried to expand on details as much as I felt was necessary for someone new to python package managers (which was myself a couple months ago).

### CODE REVIEW

File: docs/guides/projects.md [hunk 1]
```diff
@@ -1,32 +1,137 @@
...
 ```
-uv init
+.
+├── .venv
+├── pyproject.toml
+├── README.md
+├── uv.lock
+└── src
+    └── hello-world
+        └── __init__.py
```
Comment (ibraheemdev) [inline docs/guides/projects.md:32]: This is assuming that `uv init` creates the venv and lockfile. I found that this made it a little easier to explain things than having a distinction between "the things that are created by `uv init`" and "the things that are created the first time you execute `uv run`".

File: docs/guides/projects.md [hunk 2]
```diff
@@ -1,32 +1,136 @@
 # Working on projects
 
+uv can manage the development of a Python project from the ground up.
```
Comment (ibraheemdev) [inline docs/guides/projects.md:3]: Feel like we need a better tagline here.

File: docs/guides/projects.md [hunk 3]
```diff
@@ -1,32 +1,136 @@
...
+
+### `.venv`
+
+The `.venv` folder contains your project's virtual environment, a Python
+environment that is isolated from the rest of your system. This is where uv will
+install your project's dependencies.
 
+### `uv.lock`
+
+`uv.lock` is a lockfile that contains exact information about your project's
```
Comment (ibraheemdev) [inline docs/guides/projects.md:80]: Are we recommending to check this in to VCS (does that depend on application vs. library)?

Comment (zanieb) [reply docs/guides/projects.md:80]: I think it should always be checked into VCS

File: docs/guides/projects.md (hunk 2 above)
Comment (zanieb) [reply docs/guides/projects.md:3]: Maybe just define a project e.g.: uv is capable of managing projects using the `pyproject.toml` standard.

File: docs/guides/projects.md (hunk 1 above)
Comment (zanieb) [reply docs/guides/projects.md:29]: I think it's fine to create them both personally. Maybe we just want an opt-out flag?

**Review by zanieb — APPROVED** [review]
Going to merge and iterate on it after. Thanks!

**Pull request merged**